Userland/Services/RequestServer/ConnectionFromClient.cpp

```diff
@@ -73,11 +76,27 @@ size_t ConnectionFromClient::on_header_received(void* buffer, size_t size, size_
     auto* request = static_cast<ActiveRequest*>(user_data);
     size_t total_size = size * nmemb;
     auto header_line = StringView { static_cast<char const*>(buffer), total_size };
+
+    // NOTE: We need to extract the HTTP reason phrase since it can be a custom value.
+    //       Fetching infrastructure needs this value for setting the status message.
+    if (!request->reason_phrase.has_value() && header_line.starts_with("HTTP/"sv)) {
+        if (auto const space_positions = header_line.find_all(" "sv); space_positions.size() > 1) {
+            auto const second_space_offset = space_positions.at(1);
+            auto const reason_phrase_string_view = header_line.substring_view(second_space_offset + 1).trim_whitespace();
+
+            if (!reason_phrase_string_view.is_empty()) {
+                request->reason_phrase = Web::Infra::isomorphic_decode(reason_phrase_string_view.bytes());
```

Comment:
Do we really need to link against libweb just to get a stringbuilder + append_code_point in a loop?
Also cc @trflynn89 (iirc you merged that), maybe we could put that function in the header or just inline here?

Reply:
It could also be moved to LibTextCodec

Reply:
This PR was previously linking against LibTextCodec and just using a Latin1 decoder directly. Should I just revert to using that again?